src/Wordpress/CustomPostType/Booking.php

```diff
@@ -201,44 +209,49 @@ function_exists( 'wp_verify_nonce' ) &&
                 $locationId,
                 $itemId
             );
+
 	        $existingBookings =
 		        \CommonsBooking\Repository\Booking::getExistingBookings(
 			        $itemId,
 			        $locationId,
 			        $startDate,
 			        $endDate,
+                    $booking->ID,
 		        );
 
 			// Validate booking -> check if there are no existing bookings in timerange.
 			if (count($existingBookings) > 0) {
 				$requestedPostname = array_key_exists( 'cb_booking', $_REQUEST ) ? $_REQUEST['cb_booking'] : '';
+
 				// checks if it's an edit, but ignores exact start/end time
 				$isEdit = count( $existingBookings ) === 1 &&
 					array_values( $existingBookings )[0]->getPost()->post_name === $requestedPostname &&
-					array_values( $existingBookings )[0]->getPost()->post_author == get_current_user_id();
+					array_values( $existingBookings )[0]->getPost()->post_author === get_current_user_id();
 
 				if ( ( ! $isEdit || count( $existingBookings ) > 1 ) && $post_status != 'canceled' ) {
                     if ($booking) {
-                        $post_status = 'unconfirmed';
+                         $post_status = 'unconfirmed';
                         set_transient( 'commonsbooking_overlappingBooking_' . $booking->ID, $booking->ID );
                     } else {
                         throw new Exception( 'There is already a booking in this timerange.' );
                     }
 				}
+
 			}
 
+            // add internal comment if admin edited booking via frontend
```

Comment:
@datengraben @hansmorb: Hier kommme ich gerade nicht weiter. Ich will prüfen, ob der Post-Autor ungleich dem aktuellen Nutzenden ist. Dies kann ja nur der Fall der sein, wenn ein Admin die Buchung bearbeitet. Nur in diesem Fall soll dann der Booking-Admin + internal Kommentar gesetzt werden.
Irgendwie löst die Bedingung aber momentan auch aus, wenn current-user und post-author die gleiche ID haben? 
Es wird also ein direkt ein Buchungskommentar wenn nicht-Admin-User die Buchung bestätigt. 
Ich kann mir das gerade nicht erklären. Die Debug-Tests ergeben, dass beide Werte gleich sind, dennoch geht die Abfrage als true durch..
Habt ihr da ne Idee? Steh aufm Schlauch ..

Reply:
Konkret gehts hier um diesen Teil:

`      if ( $booking && $booking->post_author && $booking->post_author !== get_current_user_id() ) {
                commonsbooking_write_log('internal comment');
                $postarr['meta_input']['admin_booking_id'] = get_current_user_id();
                $internal_comment = esc_html__( 'status changed by admin user via frontend. New status: ', 'commonsbooking' ) . $post_status;
                $booking->appendToInternalComment( $internal_comment, get_current_user_id() );
            }`

Reply:
intval war wohl des Rätsels Lösung .. Seltsam, dass das unterschiedliche Typen sind?
`if ( $booking && $booking->post_author !== '' && intval( $booking->post_author ) !== intval( get_current_user_id() ) ) {`

Reply:
Stimmt, habe gerade geschaut.  https://developer.wordpress.org/reference/classes/wp_post/

```php
	/**
	 * ID of post author.
	 *
	 * A numeric string, for compatibility reasons.
	 *
	 * @since 3.5.0
	 * @var string
	 */
	public $post_author = 0;
```